File: Roboti/app/src/main/java/cz/muni/fi/umimecesky/roboti/utils/RobotDrawable.java
package cz.muni.fi.umimecesky.roboti.utils;

import android.app.Activity;
import android.content.res.Resources;
import android.graphics.drawable.Drawable;
import android.util.Log;

import java.lang.reflect.Field;
import java.security.SecureRandom;
import java.util.ArrayList;
import java.util.List;
import java.util.Random;

import cz.muni.fi.umimecesky.roboti.R;

public class RobotDrawable {

    private List<String> robotNames = new ArrayList<>();
    private Activity activity;

    private final Random RANDOM = new SecureRandom();

    public RobotDrawable(Activity activity) {

        this.activity = activity;
        Field[] drawables = R.drawable.class.getFields();
        for (Field f : drawables) {
            String fieldName = f.getName();
            if (fieldName.startsWith("robot")) {
                robotNames.add(f.getName());
            }
        }
    }


    public Drawable removeRobotDrawable() {

        Resources resources = activity.getResources();
        int num = RANDOM.nextInt(robotNames.size());
        String randomDrawableName = robotNames.remove(num);
        Log.v("num", String.valueOf(num));
        Log.v("random", randomDrawableName);
        Log.v("list", String.valueOf(robotNames));
        final int resourceId = resources.getIdentifier(randomDrawableName, "drawable",
                activity.getPackageName());
        return resources.getDrawable(resourceId);
    }
}

File: Roboti/app/src/main/java/cz/muni/fi/umimecesky/game/robots/RobotDrawable.kt
package cz.muni.fi.umimecesky.game.robots

import android.content.Context
import android.graphics.drawable.Drawable
import android.support.v4.content.ContextCompat
import cz.muni.fi.umimecesky.R

object RobotDrawable {

    private val robotNames : List<String> =
            R.drawable::class.java.fields
                    .map { it.name }
                    .filter { it.startsWith("robot") }

    private val uniqueRandom: UniqueRandom = UniqueRandom(robotNames.size)

    fun getRobotDrawable(context: Context): Drawable {

        val randomInt = uniqueRandom.next()
        val randomDrawableName = robotNames[randomInt]
        val resourceId = context.resources.getIdentifier(randomDrawableName, "drawable",
                context.packageName)
        return ContextCompat.getDrawable(context, resourceId)!!
    }
}
